registry.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+const namespace = "geth"
+
+type registry struct {
+	c *collector
+	*prometheus.Registry
+}
+
+func newRegistry(ipcPath string, rawFilters []string) *registry {
+	return &registry{
+		c:        newCollector(ipcPath, rawFilters),
+		Registry: prometheus.NewRegistry(),
+	}
+}
+
+// Gather implements the prometheus.Registerer interface
+func (r *registry) Gather() ([]*dto.MetricFamily, error) {
+	var families []*dto.MetricFamily
+
+	fs, err := r.Registry.Gather()
+	if err != nil {
+		log.Println(err)
```

Comment:
I followed the `Gatherer` interface docs that say that the response can be inclomplete in case of errors

https://github.com/prometheus/client_golang/blob/master/prometheus/registry.go#L126